### DISCUSSION TITLE
Length of beatmaps don't change when DT/HT is applied

### DISCUSSION THREAD

**Discussion by Flamiii** [ppy/osu#15265] [Ideas] [answered]
I'm sorry in advance if what I said in this post sounds super hostile and/or aggressive. I have an immense level of respect for everyone on the osu!dev team and I just want to see the game improve for the better. 

I noticed there was an inconsistency with the way osu!stable and osu!lazer handle the length of a beatmap when DT/HT (and Wind Up / Wind Down) is applied and I just wanted to bring it up.

![osu!lazer length not changing](https://user-images.githubusercontent.com/74088315/138584529-3af65d76-0e8c-49a5-8714-0bb7757ae36a.png)

Even if this is intentional, I personally just find it weird for the length to not change in this case since the beatmap does actually take less physical time to play through. For example, when using DT in osu!lazer, instead of the length of the beatmap being 2/3 of its original value like it is in osu!stable, each second/minute progresses 150% faster, which in my opinion has the potential to be extremely misleading. A map like The Unforgiving which is normally 52 minutes and 54 seconds long becomes 35 minutes and 16 seconds long when DT is applied, which is substantially different than the original value that is displayed in song select.

If you play both of the following videos at the same time, you will notice that one of them progresses through the beatmap much faster than the other, and this is because the first video is with no mods, while the second video is with DT. There is nothing inherently wrong with this being the case, but I do take issue with the fact that time is seemingly 'progressing faster' when DT is applied. Note that this 'time progressing faster' thing also occurs during breaks where there is a countdown on screen.

https://user-images.githubusercontent.com/74088315/138585471-34b522e6-f577-4fc5-8e97-9648ce1782fd.mp4

During the first video (~30 seconds long in actual seconds), ~30 seconds have passed in the game, which makes sense.

https://user-images.githubusercontent.com/74088315/138585474-26443c5c-6cd1-4997-af34-93d69174b49e.mp4

During the second video (~30 seconds long in actual seconds), ~47 'seconds' have passed in the game (realistically should be closer to 45 seconds but that's my fault), which doesn't really make sense to me. If the DT mod is being used, the length of the beatmap should be reduced to 2/3 of its original value like it is in osu!stable, instead of making time progress 150% faster.

As a player, I just feel like this sort of thing makes the length statistic of any beatmap pretty much meaningless whenever a mod like DT or HT is used. In my opinion, changing how long a second or a minute lasts based on the mods that you're using is a pretty unintuitive way to handle this. The length of a beatmap is something that I can understand because I know that the length is displayed in actual units of time. However, when using DT or HT, this is no longer the case, and the player is left to handle the calculations of how long a map is in actual minutes and seconds themselves. In my honest opinion, osu!stable handles this a lot better than osu!lazer currently does, and I think it would make much more sense to players if it was handled in that same way.

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/issues/10580
